src/main/java/com/microsoft/azure/functions/worker/chain/SdkTypeMiddleware.java

```
@@ -53,16 +53,19 @@ public void invoke(MiddlewareContext context, MiddlewareChain chain) throws Exce
             WorkerObjectCache<CacheKey> cache = execCtx.getCache();
 
             for (SdkTypeMetaData metaData : this.sdkTypesMetaData) {
+                Parameter param = metaData.getParam();
+                ParamBindInfo paramBindInfo = new ParamBindInfo(param);
                 Set<String> requiredKeys = metaData.getRequiredFields();
 
                 for (String key : requiredKeys) {
-                    Object val = dataStore.getDataByName(key, String.class)
+                    Object val = dataStore.getDataByNameFromInputSource(key, String.class, paramBindInfo.getName())
                             .map(b -> b.getValue())
                             .orElseThrow(() -> new IllegalArgumentException("Missing " + key));
```

Missing what?

Can you enrich the exception messaging

---

If key is optional do you need to throw the argument exception? Otherwise you can use replace orElseThrow with following

```
    .ifPresent(val -> metaData.setFieldValue(key, val));
```